### DISCUSSION TITLE
Can we get the GUID if mac address is known - iOS

### DISCUSSION THREAD

**Discussion by Dinesh-Fernando** [dotnet-bluetooth-le/dotnet-bluetooth-le#733] [Q&A] [answered]
Is there a way to get the GUID if the Mac address is known on iOS?
I know this is possible with Android using the following

Guid.Parse($"00000000-0000-0000-0000-{macAddress.Replace(":", "")}")

However iOS it seems like the GUID is quite different.

eg: 46d25d75-b17c-ef6b-4dd6-f0af8054d9e2

Thanks
Dinesh

**Comment by AskBojesen** [comment, marked as answer]
I don't think so. 

The Id in the Apple implementation does NOT contain the BLE Address which is allways 48 bit = 6 bytes = 12 hex chars , but something else.

Eg... 
https://developer.apple.com/forums/thread/8442
https://stackoverflow.com/questions/18973098/get-mac-address-of-bluetooth-low-energy-peripheral-in-ios
